senderworker/src/main/scala/com/gu/notifications/ec2worker/SenderWorker.scala

```diff
@@ -2,60 +2,60 @@ package com.gu.notifications.ec2worker
+import com.amazonaws.services.lambda.runtime.Context
+import com.amazonaws.services.sqs.{AmazonSQSAsyncClientBuilder, AmazonSQSClientBuilder}
+import com.amazonaws.services.sqs.model.{Message, ReceiveMessageRequest}
 import com.gu.notifications.worker.IOSSender
 import com.gu.notifications.worker.AndroidSender
+import com.gu.notifications.worker.utils.Aws.credentialsProvider
+
+import scala.concurrent.Future
 
 object SenderWorker extends App {
 
   implicit val logger: Logger = LoggerFactory.getLogger(this.getClass)
 
-  val notification = BreakingNewsNotification(
-    id = UUID.randomUUID(),
-    `type` = NotificationType.BreakingNews,
-    title = Some("French president Francois Hollande says killers of Normandy priest claimed to be from Islamic State"),
-    message = Some("French president Francois Hollande says killers of Normandy priest claimed to be from Islamic State"),
-    thumbnailUrl = None,
-    sender = "matt.wells@guardian.co.uk",
-    link = Link.Internal(
-      "world/2016/jul/26/men-hostages-french-church-police-normandy-saint-etienne-du-rouvray",
-      Some("https://gu.com/p/4p7xt"),
-      GITContent,None,
-    ),
-    imageUrl = None,
-    importance = Importance.Major,
-    topic = List(Topic(TopicTypes.Breaking, "uk"), Topic(TopicTypes.Breaking, "us"), Topic(TopicTypes.Breaking, "au"), Topic(TopicTypes.Breaking, "international")),
-    dryRun = Some(true)
-  )
-
-  val tokens = ChunkedTokens(
-    notification = notification,
-    range = ShardRange(0, 1),
-    tokens = List("token")
-  )
-
-  val sqsEvent: SQSEvent = {
-    val event = new SQSEvent()
-    val sqsMessage = new SQSMessage()
-    sqsMessage.setBody(Json.stringify(Json.toJson(tokens)))
-    sqsMessage.setAttributes((Map("SentTimestamp" -> s"${Instant.now.toEpochMilli}").asJava))
-    event.setRecords(List(sqsMessage).asJava)
-    event
+  val sqsClient = AmazonSQSAsyncClientBuilder.standard()
+    .withCredentials(credentialsProvider)
+    .withRegion("eu-west-1")
+    .build()
+
+  def pollQueue(queue: String, handleChunkTokens: (SQSEvent, Context) => Unit): Future[Unit] = Future {
+    logger.info(s"About to poll queue - $queue")
+    val receiveMessages = sqsClient.receiveMessage(new ReceiveMessageRequest().withQueueUrl(queue).withWaitTimeSeconds(20)).getMessages.asScala.toList
+
+    def parsedReceivedMessages(message: List[Message]): SQSEvent = {
+      val parsedMessages = message.map(m => {
+        val sqsMessage = new SQSMessage()
+        sqsMessage.setBody(m.getBody)
+        sqsMessage.setAttributes(m.getAttributes)
+        sqsMessage.setReceiptHandle(m.getReceiptHandle)
+        sqsMessage
+      })
+
+      val event = new SQSEvent()
+      event.setRecords(parsedMessages.asJava)
+      event
+    }
+
+    val parsedEvent: SQSEvent = parsedReceivedMessages(receiveMessages)
+
+    handleChunkTokens(parsedEvent, null)
+
+    parsedEvent.getRecords.forEach(m => {
```

Comment:
At the moment `handleChunkTokens` returns `Unit` and the method itself handles all failures resulting from processing. Without changing some of the internal logic of this method (used by both lambdas and ec2) we didn't think we could conditionally delete messages from the source queue and/or put onto a dlq.

We decided for now that, as long as the ec2 service has started correctly and has begun polling, we will delete any messages we receive.

We should check what happens if any of the messages fail to send (I _thought_ we logged errors from handleChunkTokens and also triggered alerts)